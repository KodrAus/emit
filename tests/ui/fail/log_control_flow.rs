#![feature(stmt_expr_attributes, proc_macro_hygiene)]
#![deny(warnings)]

fn main() {
    emit::info!("Text and {a: return 42}");
}
