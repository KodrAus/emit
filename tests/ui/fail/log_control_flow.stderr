error[E0308]: mismatched types
 --> fail/log_control_flow.rs:5:28
  |
4 | fn main() {
  |           - expected `()` because of default return type
5 |     emit::info!("Text and {a: return 42}");
  |                            ^^^^^^^^^^^^ expected `()`, found integer

error: unreachable call
 --> fail/log_control_flow.rs:5:28
  |
5 |     emit::info!("Text and {a: return 42}");
  |                            ^^^^^^^^^^^^
  |                            |
  |                            unreachable call
  |                            any code following this expression is unreachable
  |
note: the lint level is defined here
 --> fail/log_control_flow.rs:2:9
  |
2 | #![deny(warnings)]
  |         ^^^^^^^^
  = note: `#[deny(unreachable_code)]` implied by `#[deny(warnings)]`
  = note: this error originates in the macro `emit::ct::__private_capture` (in Nightly builds, run with -Z macro-backtrace for more info)
